nucypher/policy/conditions/lingo.py

```diff
@@ -122,37 +122,30 @@ def _sanitize_value(self, value):
 
     def _process_data(self, data: Any) -> Any:
         """
-        If a key is specified, return the value at that key in the data if data is a dict or list-like.
+        If an index is specified, return the value at that index in the data if data is list-like.
         Otherwise, return the data.
         """
         processed_data = data
-        if self.key is not None:
-            if isinstance(data, dict):
+        if self.index is not None:
+            if isinstance(self.index, int) and isinstance(data, (list, tuple)):
```

Comment:
Is this `isinstance(self.index, int)` already checked by this point and therefore not needed ? It's from the `ReturnValueTest` object which already checks that index is an `int` in the constructor.

Reply:
completely, but I actually like having it there because I feel it makes the code more readable (it's really clear what that line is trying to achieve)

removing it would make the code more efficient, but it's extremely minor